tests/integration-tests/tests/storage/test_shared_home.py

```diff
@@ -11,36 +11,104 @@
 # See the License for the specific language governing permissions and limitations under the License.
 import logging
 
+import boto3
 import pytest
+from assertpy import assert_that
 from remote_command_executor import RemoteCommandExecutor
 
 from tests.storage.storage_common import (
+    check_fsx,
+    create_fsx_ontap,
+    create_fsx_open_zfs,
     test_directory_correctly_shared_between_ln_and_hn,
     test_efs_correctly_mounted,
     verify_directory_correctly_shared,
 )
 
 
+@pytest.mark.parametrize(
+    "storage_type",
+    ["Efs", "FsxLustre", "FsxOpenZfs", "FsxOntap", "Ebs"],
+    # TODO full test ["Efs", "FsxLustre", "FsxOpenZfs", "FsxOntap", "FileCache", "Ebs"]
+)
 @pytest.mark.usefixtures("os", "scheduler", "instance")
 def test_shared_home(
-    region, scheduler, pcluster_config_reader, clusters_factory, vpc_stack, scheduler_commands_factory
+    region,
+    scheduler,
+    pcluster_config_reader,
+    create_file_cache,
+    vpc_stack,
+    scheduler_commands_factory,
+    storage_type,
+    fsx_factory,
+    svm_factory,
+    open_zfs_volume_factory,
+    s3_bucket_factory,
+    test_datadir,
+    clusters_factory,
 ):
-    """Verify the internal shared storage fs is available when set to Efs"""
+    """Verify the shared /home storage fs is available when set"""
     mount_dir = "/home"
-    cluster_config = pcluster_config_reader(mount_dir=mount_dir)
+    bucket_name = None
+    if storage_type == "FileCache":
+        bucket_name = s3_bucket_factory()
+        bucket = boto3.resource("s3", region_name=region).Bucket(bucket_name)
+        bucket.upload_file(str(test_datadir / "s3_test_file"), "s3_test_file")
+    file_cache_path = "/slurm/" if storage_type == "FileCache" else None
+    if storage_type == "FsxOpenZfs":
+        fsx_open_zfs_root_volume_id = create_fsx_open_zfs(fsx_factory, num=1)[0]
+        fsx_open_zfs_volume_id = open_zfs_volume_factory(fsx_open_zfs_root_volume_id, num_volumes=1)[0]
+        cluster_config = pcluster_config_reader(
+            mount_dir=mount_dir, storage_type=storage_type, volume_id=fsx_open_zfs_volume_id
+        )
+    elif storage_type == "FsxOntap":
+        fsx_ontap_fs_id = create_fsx_ontap(fsx_factory, num=1)[0]
+        fsx_on_tap_volume_id = svm_factory(fsx_ontap_fs_id, num_volumes=1)[0]
+        cluster_config = pcluster_config_reader(
+            mount_dir=mount_dir, storage_type=storage_type, volume_id=fsx_on_tap_volume_id
+        )
+    elif storage_type == "FileCache":
```

Comment:
For the next PR when we add FileCache after the performance is improves, why do we have two different if conditions for FileCache?